### ISSUE TITLE
dif/2 loops, and cannot be interrupted

### ISSUE THREAD

**Issue by UWN** [trealla-prolog/trealla#185]
[Item#18](http://www.complang.tuwien.ac.at/ulrich/trealla-prolog/Item#18)

**Comment by UWN** [comment]
One clarification. The problem is not infinite trees, but already:

```
?- dif([],A), A = [[]|B].
   A = [[]|B], dif([],[[]|B]), unexpected.
   A = [[]|B]. % expected, but not found
```

**Comment by infradig** [comment]
Not fixed, sorry about that. Scryer is wrong too,

**Comment by UWN** [comment]
Right, and [also ECLiPSe](http://www.complang.tuwien.ac.at/ulrich/iso-prolog/dif#12) has such redundant constraints. The best solution to this would be something along [this suggested approach](https://github.com/mthom/scryer-prolog/issues/135#issuecomment-488348886). And, to be fair, all three at least make diagnosing this weakness possible, whereas IV, IF, and B simply never show pending constraints, so it is not clear what is happening there. (It's possible to find this out by putting such queries into a loop and then check if GC is able to recover it, but then this is slightly out of scope of 13211-1.)

**Comment by UWN** [comment]
Closed with followup #208
